### ISSUE TITLE
Hue Entertainment won't resume after Kodi paused/resumed

### ISSUE THREAD

**Issue by AnthraxLeprosyPi** [hyperion-project/hyperion.ng#1059]
Hi everybody,

it's been a while...

I had my first days off in ages and figured I should update my hyperion setup to the latest and greatest ;-)

All seemed to work well until I paused a video in Kodi and resumed shortly afterwards.

I have three instances running (1. LED-Ambilight, 2. Hue Entertainment and 3. Nanoleaf) and they all behave differently:

1. LED-Ambilight: 
- while the video is paused the LEDs still output accoring to what is shown (still image from video and player ui)
- after playback resumed the LEDs continue their output as well 👍 

2. Hue Entertainment:
- when the video is paused the Hue lights shut off shortly after
- after playback resumed the LEDs wont resume to show the hyperion output and stay off 👎 

3. Nanoleaf:
- when the video is paused the nanoleafs shut off shortly after
- after playback resumed the LEDs resume to show the hyperion output as expected 👍

TL;DR: 
The hue instance won't resume after video was paused and resumed.
Is there any new setting that might cause this or is this a bug?

Log output from web interface (#conf_logging):
https://paste.osmc.tv/cericojehe.coffee

Ideally the behavior could be set by the user via options (i.e: keep putting out even when video is paused / turn LED-Device(s) OFF / Choose paused color or effect)

Cheers
Anthrax

**Comment by Lord-Grey** [comment]
Closed as duplicate to #930.

@AnthraxLeprosyPi Suggest you disable „switchoff on black“ at the Hue Device.
Then the device should behave as the other two.

**Comment by AnthraxLeprosyPi** [comment]
Hi @Lord-Grey,

thanx for the info - I'll as soon as I'm home.

But shouldn't it be able resume anyway though?

Anyways. Cheers,
Anthrax

**Comment by Lord-Grey** [comment]
> "But shouldn't it be able resume anyway though?"

Yes,if the device switches off itself, then it should be able switching it on itself too. 
As this is not working currently, issue 930 is in place...

This was working before, but I did updates to the general flow for the LED-Devices, why this functionality requires fixing (or better removed from the device).

**Comment by AnthraxLeprosyPi** [comment]
Thanx for the insight @Lord-Grey.

Unfortunately your suggested workaround didn't work 😕.

These are my settings (which I have never touched or at least not in a very long time as far as I know of) - are these allright?

Keep it up!
Anthrax

**Comment by Lord-Grey** [comment]
@AnthraxLeprosyPi Sorry, I re-checked to code and you are right, switchOffonBlack is not considered when Entertainment API is on.

If you compile yourself, you might have it configurable by adding the _switchOffOnBlack condition.

```
bool LedDevicePhilipsHue::noSignalDetection()
{
	if( _allLightsBlack && _switchOffOnBlack)
	{
```

To deactivate the "Switch off on black" you need temporarily disable Entertainment API
Then disable ""Switch off on black"
![image](https://user-images.githubusercontent.com/48840279/96998168-4a5c2400-1533-11eb-88dc-e6d4f9141c5b.png)

and enable Entertainment API again. Then save.

In the debug log you should see
```
LedDevicePhilipsHue.cpp:417:log() | Off on Black                  : 0
...
LedDevicePhilipsHue.cpp:417:log() | Use Hue Entertainment API     : 1
```
That might provide a quick workaround, until it will be fixed in general.
If this is working, please report back. 
I might check this in as an interim solution, until we get it fixed in general.
Having this behaviour configurable would anyway beneficial, if I am not mistaken....

**Comment by AnthraxLeprosyPi** [comment]
@Lord-Grey,

I tried your workaround and was able to deactivate "Switch off on black" but it didn't seem to have changed anything. :sad:
The lights won't resume once playback is resumed.

anyways thanksfor your support,
Anthrax

**Comment by Lord-Grey** [comment]
@AnthraxLeprosyPi With the workaround, the lights should have not even switched-off before....
i.e. there would be no need to resume....

Would be maybe good to understand how you run the stop playback and resume scenario.

**Comment by AnthraxLeprosyPi** [comment]
@Lord-Grey ,

well, after I pause the video within Kodi there is a slight delay (maybe 2 seconds) and the hues and nanoleaf turn off/black. The the TV-Ambilight from the main instance keeps showing the paused image.

As soon as I resume playback the the nanoleaf resume perfectly but the hues won't - and yes I understand that the hues shouldn't turn off with this setting but this is what they'll do ☹️ ...

**Comment by AnthraxLeprosyPi** [comment]
> @AnthraxLeprosyPi Sorry, I re-checked to code and you are right, switchOffonBlack is not considered when Entertainment API is on.
> 
> If you compile yourself, you might have it configurable by adding the _switchOffOnBlack condition.
> 
> ```
> bool LedDevicePhilipsHue::noSignalDetection()
> {
> 	if( _allLightsBlack && _switchOffOnBlack)
> 	{
> ```
> 
> To deactivate the "Switch off on black" you need temporarily disable Entertainment API
> Then disable ""Switch off on black"
> ![image](https://user-images.githubusercontent.com/48840279/96998168-4a5c2400-1533-11eb-88dc-e6d4f9141c5b.png)
> 
> and enable Entertainment API again. Then save.
> 
> In the debug log you should see
> 
> ```
> LedDevicePhilipsHue.cpp:417:log() | Off on Black                  : 0
> ...
> LedDevicePhilipsHue.cpp:417:log() | Use Hue Entertainment API     : 1
> ```
> 
> That might provide a quick workaround, until it will be fixed in general.
> If this is working, please report back.
> I might check this in as an interim solution, until we get it fixed in general.
> Having this behaviour configurable would anyway beneficial, if I am not mistaken....

Hi Lord-Grey,

I've just read your instructions again and stumbled upon "If you compile yourself, you might have it configurable by adding the _switchOffOnBlack condition." - Did you mean I HAVE TO COMPILE myself and then the workaround should work or should it work just by simply tricking the ui as described?

Man I feel stupid now ;-)

Anthrax

**Comment by Lord-Grey** [comment]
@AnthraxLeprosyPi Sorry, looks like I have not expressed myself clear enough.
To overcome the issue a code change is required mandatorily.

Either via the workaround outlined above or with a general fix.

If you do not want to wait for a general fix, you could do the code change yourself and this mean you need to compile Hyperion yourself to get the workaround in place.

**Comment by AnthraxLeprosyPi** [comment]
> @AnthraxLeprosyPi Sorry, looks like I have not expressed myself clear enough.
> To overcome the issue a code change is required mandatorily.
> 
> Either via the workaround outlined above or with a general fix.
> 
> If you do not want to wait for a general fix, you could do the code change yourself and this mean you need to compile Hyperion yourself to get the workaround in place.

Hi @Lord-Grey, 

yeah already made the code change and rebuilt, and to no one's (ok, my) surprise it works ;-)

So, I can live with this workaround.

THANK YOU
Anthrax

**Comment by Lord-Grey** [comment]
@AnthraxLeprosyPi Good to hear.

> and to no one's (ok, my) surprise it works 

Luckily not to my surprise (as I tested it before...)  :)
